lib/trento.pm

```diff
@@ -0,0 +1,369 @@
+
+# Lib internal constants
+use constant TRENTO_AZ_PREFIX => 'openqa-trento';
+# Parameter 'registry_name' must conform to
+# the following pattern: '^[a-zA-Z0-9]*$'.
+# Azure does not support dash or underscore in ACR name
+use constant TRENTO_AZ_ACR_PREFIX => 'openqatrentoacr';
+use constant PODMAN_PULL_LOG => '/tmp/podman_pull.log';
+use constant CYPRESS_IMAGE_TAG => 'goofy';
+
+=head1 DESCRIPTION 
+
+Package with common methods and default or constant  values for Trento tests
+
+=head2 Methods
+
+=head3 get_resource_group
+
+Return a string to be used as cloud resource group.
+It contains the JobId
+=cut
+sub get_resource_group {
+    my $job_id = get_current_job_id();
+    return TRENTO_AZ_PREFIX . "-rg-$job_id";
+}
+
+=head3 get_vm_name
+
+Return a string to be used as cloud VM name.
+It contains the JobId
+=cut
+sub get_vm_name {
+    my $job_id = get_current_job_id();
+    return TRENTO_AZ_PREFIX . "-vm-$job_id";
+}
+
+=head3 get_acr_name
+
+Return a string to be used as cloud ACR name.
+It contains the JobId
+=cut
+sub get_acr_name {
+    return TRENTO_AZ_ACR_PREFIX . get_current_job_id();
+}
+
+=head3 az_get_vm_ip
+
+Return the running VM public IP
+=cut
+sub az_get_vm_ip {
+    my $az_cmd = 'az vm show -d ' .
+      '-g ' . get_resource_group() . ' ' .
+      '-n ' . get_vm_name() . ' ' .
+      '--query "publicIps" ' .
+      '-o tsv';
+    return script_output($az_cmd, 180);
+}
+
+=head3 az_delete_group
+
+Delete the resource group associated to this JobID and all its content
+=cut
+sub az_delete_group {
+    script_run('echo "Delete all resources"');
+    my $az_cmd = 'az group delete ' .
+      '--resource-group ' . get_resource_group() .
+      ' --yes';
+    script_retry($az_cmd, timeout => 600, retry => 5, delay => 60);
+}
+
+=head3 az_vm_ssh_cmd
+
+Compose I<ssh> command for remote execution on the VM machine.
+The function optionally accept the VM public IP.
+If not provided, the IP is calculated on the fly with an I<az> query,
+take care that is a time consuming I<az> query.
+
+=over 2
+
+=item B<CMD_ARG> - String of the command to be executed remotely
+
+=item B<VM_IP_ARG> - Public IP of the remote machine where to execute the command
+
+=back
+=cut
+sub az_vm_ssh_cmd {
+    my ($self, $cmd_arg, $vm_ip_arg) = @_;
+    record_info('REMOTE', "cmd:$cmd_arg");
+
+    # undef comparision operator
+    my $vm_ip = $vm_ip_arg // az_get_vm_ip();
+    return 'ssh' .
+      ' -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR' .
+      ' -i ' . SSH_KEY . ' ' .
+      VM_USER . '@' . $vm_ip .
+      ' -- ' . $cmd_arg;
+}
+
+=head3 k8s_logs
+
+Get all relevant info out from the cluster
+
+=over 2
+
+=item B<CMD_ARG> - String of the command to be executed remotely
+
+=item B<VM_IP_ARG> - Public IP of the remote machine where to execute the command
+
+=back
+=cut
+sub k8s_logs {
+    my $self = shift;
+    my (@pods_list) = @_;
+    my $machine_ip = az_get_vm_ip;
+
+    if ($machine_ip) {
```

Comment:
I think you can re-write this as `return unless ($machine_ip);` since there is no else in this if-block and you would remove a full 4-spaces indentation below as well as some brackets. It would improve readibility IMO.